### PR TITLE
[COLLABORATION] Added architecture for rename events

### PR DESCRIPTION
**Pull request by hkirat**
In reference to [#2353](https://github.com/mozilla/thimble.mozilla.org/issues/2353)

### CODE REVIEW

File: src/editor/Collaboration.js [hunk 1]
```diff
@@ -41,6 +43,12 @@ define(function (require, exports, module) {
 
             _webrtc.connection.on('message', _handleMessage);
         });
+
+        FileSystemEntry.on("rename", function(event, oldPath, newPath) {
+            var relOldPath = oldPath.replace(StartupState.project("root"), "");
```
Comment (gideonthomas) [inline src/editor/Collaboration.js]: use Filer.path instead

File: src/editor/Collaboration.js [hunk 2]
```diff
@@ -55,6 +63,11 @@ define(function (require, exports, module) {
             case "codemirror-change":
                 _handleCodemirrorChange(msg.payload);
                 break;
+            case "file-rename":
+                var oldPath = StartupState.project("root") + msg.payload.oldPath;
```
Comment (gideonthomas) [inline src/editor/Collaboration.js]: same as abve

File: src/editor/Collaboration.js [hunk 3]
```diff
@@ -1,7 +1,11 @@
 define(function (require, exports, module) {
     "use strict";
 
+    var FileSystem = require("filesystem/FileSystem");
+    var StartupState    = require("bramble/StartupState");
     var SimpleWebRTC    = require("simplewebrtc");
+    var BracketsFiler   = require("filesystem/impls/filer/BracketsFiler");
```
Comment (gideonthomas) [inline src/editor/Collaboration.js]: hmm I think I was wrong. You would use:
```js
var Path = require("filesystem/impls/filer/FilerUtils").Path;
```

File: src/editor/Collaboration.js [hunk 4]
```diff
@@ -1,7 +1,11 @@
 define(function (require, exports, module) {
     "use strict";
 
+    var FileSystem = require("filesystem/FileSystem");
```
Comment (gideonthomas) [inline src/editor/Collaboration.js]: fix your indentation here

File: src/editor/Collaboration.js [hunk 5]
```diff
@@ -39,6 +43,11 @@ define(function (require, exports, module) {
 
         _pending = []; // pending clients that need to be initialized.
         _changing = false;
+        FileSystem.on("rename", function(event, oldPath, newPath) {
```
Comment (gideonthomas) [inline src/editor/Collaboration.js]: add a blank line above

File: src/editor/Collaboration.js [hunk 6]
```diff
@@ -39,6 +43,11 @@ define(function (require, exports, module) {
...
+            var relOldPath = oldPath.replace(StartupState.project("root"), "");
```
Comment (gideonthomas) [inline src/editor/Collaboration.js]: you're still using string replace here

File: src/editor/Collaboration.js [hunk 7]
```diff
@@ -53,6 +62,11 @@ define(function (require, exports, module) {
             case "codemirror-change":
                 _handleCodemirrorChange(msg.payload);
                 break;
+            case "file-rename":
+                var oldPath = Path.relative(msg.payload.oldPath, root);
```
Comment (gideonthomas) [inline src/editor/Collaboration.js]: we use msg.payload a lot here. Let's reduce duplication:
```js
var payload = msg.payload;
var oldPath, newPath;

switch(msg.type) {
    case ...;
    case "file-rename":
        oldPath = Path.relative(payload.oldPath, root);
        newPath = Path.relative(payload.newPath, root);
        console.log("Renamed ", oldPath, " to ", newPath);
        break;
    case ...;
}
```
and change every reference to the `payload` var to use the one you create above

**Comment by hkirat** [comment]
Implemented review comments

**Pull request merged**